src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -429,17 +429,21 @@ def _invalid_user_lockfile_error(
         yield (
             f"- The targets use interpreter constraints (`{user_interpreter_constraints}`) that "
             "are not compatible with those used to generate the lockfile "
-            f"(`{metadata.valid_for_interpreter_constraints}`). You can fix this by either "
-            f"adjusting your targets to use different interpreter constraints "
-            f"({doc_url('python-interpreter-compatibility')}) or by generating the lockfile with "
-            f"different interpreter constraints by setting the option "
-            f"`[python].resolves_to_interpreter_constraints`, then running `generate-lockfiles`.\n\n"
+            f"(`{metadata.valid_for_interpreter_constraints}`).\nThe lockfile's interpreter "
+            f"constraints are set by the option `[python].resolves_to_interpreter_constraints`, "
+            f"which determines how the lockfile is generated. Note that that option does not "
+            f"impact your own code's interpreter constraints in any way, which must be set via "
```

Comment:
It does impact it, in the sense that your code's requirements must be a subset of the lockfile's requirements, right?